cmd/promoted-image-governor/main.go

```diff
@@ -262,6 +272,47 @@ func isMirroredFromOCP(tag api.ImageStreamTagReference, refs []ImageStreamRef) b
 	return false
 }
 
+type githubClient interface {
+	GetRef(org, repo, ref string) (string, error)
+}
+
+// checkImageStreamTags checks if promotedTags are built with the current head of org/repo/branch
+func checkImageStreamTags(ctx context.Context, client ctrlruntimeclient.Client, promotedTags map[api.ImageStreamTagReference]api.Metadata, gitHubClient githubClient) []error {
+	var errs []error
+	for promotedTag, metadata := range promotedTags {
+		ist := &imagev1.ImageStreamTag{}
+		istName := fmt.Sprintf("%s:%s", promotedTag.Name, promotedTag.Tag)
+		if err := client.Get(ctx, ctrlruntimeclient.ObjectKey{Namespace: promotedTag.Namespace, Name: istName}, ist); err != nil {
+			if kerrors.IsNotFound(err) {
+				// The tag has not been promoted yet
+				logrus.WithError(err).Warnf("failed to get promoted isTag %s in namespace %s", istName, promotedTag.Namespace)
```

Comment:
put the comment into the error message, and consider logging nothing here since this is an expected state

Reply:
Removed.